### ISSUE TITLE
Get Neural Network running

### ISSUE THREAD

**Issue by chrisjbaik** [paivaspol/EECS583#3]
Currently the training isn't working. Two things are suspect:
- [x] Test input is bad. Try doing a test input with randomly generated numbers from 0-9 in the matrix, and labels for each row being the `sum(row) % 10`, and then put it in a one-hot vector form.
- [x] Cross-entropy at intermediate steps seems to be producing `NaN`. how come?

**Comment by arquinn** [comment]
well... so Cross-entropy, according to wikipedia, is a sum of a probability times a log
SO.. log(x) = NaN when x < 0 right?

**Comment by chrisjbaik** [comment]
@arquinn you're absolutely right. I'm gonna take a cue from this page:

http://stackoverflow.com/questions/33712178/tensorflow-nan-bug

It's still not really changing much though; have to figure out what's going wrong.

**Comment by chrisjbaik** [comment]
This problem is fixed, but we're still not getting super awesome results. :(

@zainahamid @arquinn @paivaspol
